File: Data Structures & Algorithms/valid-palindrome-ii/submission-1.py
class Solution:
    def validPalindrome(self, s: str) -> bool:
        if len(s) == 1: return True

        if self.isPal(s): return True

        res = []
        for i in range(len(s)):
            l = s[:i] + s[i+1:]
            res.append(self.isPal(l))

        print(res)


        if True in res:
            return True
        else:
            return False


    def isPal(self, w) :

        l =0
        r = len(w) -1

        while l<r:
            if w[l] != w[r]: 
                return False

            r-=1
            l+=1

        return True

File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-1.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int l = 0;
        int subsum = 0;
        int res = INT_MAX;

        for (int r=0; r<nums.size();r++){
            subsum += nums[r];
            while (subsum >= target){
                res = std::min(res, (r-l+1));
                subsum -= nums[l];
                l++;
            }
        }
        return res == INT_MAX ? 0: res;
    }
};